src/main/java/com/iexec/core/task/TaskService.java

```diff
@@ -432,6 +433,64 @@ private void initializedOrRunning2ContributionTimeout(Task task) {
         }
     }
 
+    /**
+     * When all workers have failed to run the task, we send the new status as soon as possible.
+     * It avoids a state within which we know the task won't be updated anymore, but we wait for a timeout.
+     * <br>
+     * We consider that all workers are in a `RUNNING_FAILED` status when:
+     * <ol>
+     *     <li>All alive workers have tried to run the task;</li>
+     *     <li>All alive workers that have tried to run the task have failed.</li>
+     * </ol>
+     *
+     * @param task Task to check and to make become {@link TaskStatus#RUNNING_FAILED}.
+     */
+    private void running2RunningFailed(Task task) {
+        boolean isRunningTask = task.getCurrentStatus().equals(RUNNING);
+        if (!isRunningTask) {
+            return;
+        }
+
+        final ReplicatesList replicatesList = replicatesService.getReplicatesList(task.getChainTaskId()).orElseThrow();
+        final List<Worker> aliveWorkers = workerService.getAliveWorkers();
+
+        final List<Optional<Replicate>> replicatesOfAliveWorkers = aliveWorkers
+                .stream()
+                .map(Worker::getWalletAddress)
+                .map(replicatesList::getReplicateOfWorker)
```

Comment:
The list of optionals doesn't add value, you can filter optional to get a plain list, then compare its size to the alive workers list.

Reply:
You're right. That's fixed :)